frappe/integrations/doctype/google_contacts/google_contacts.py

```diff
@@ -136,24 +136,27 @@ def sync(g_contact=None):
 
 				for name in connection.get("names"):
 					if name.get("metadata").get("primary"):
+						contact = frappe.get_doc({
+							"doctype": "Contact",
+							"salutation": name.get("honorificPrefix") if name.get("honorificPrefix") else "",
+							"first_name": name.get("givenName") if name.get("givenName") else "",
+							"middle_name": name.get("middleName") if name.get("middleName") else "",
+							"last_name": name.get("familyName") if name.get("familyName") else "",
+							"designation": get_indexed_value(connection.get("organizations"), 0, "title"),
+							"source": "Google Contacts",
+							"google_contacts_description": get_indexed_value(connection.get("organizations"), 0, "name")
+						})
+
 						if connection.get("emailAddresses"):
 							for email in connection.get("emailAddresses"):
-								if not frappe.db.exists("Contact", {"email_id": email.get("value")}):
-									contacts_updated += 1
-
-									frappe.get_doc({
-										"doctype": "Contact",
-										"salutation": name.get("honorificPrefix") if name.get("honorificPrefix") else "",
-										"first_name": name.get("givenName") if name.get("givenName") else "",
-										"middle_name": name.get("middleName") if name.get("middleName") else "",
-										"last_name": name.get("familyName") if name.get("familyName") else "",
-										"email_id": email.get("value") if email.get("value") else "",
-										"designation": get_indexed_value(connection.get("organizations"), 0, "title"),
-										"phone": get_indexed_value(connection.get("phoneNumbers"), 0, "value"),
-										"mobile_no": get_indexed_value(connection.get("phoneNumbers"), 1, "value"),
-										"source": "Google Contacts",
-										"google_contacts_description": get_indexed_value(connection.get("organizations"), 0, "name")
-									}).insert(ignore_permissions=True)
+								contact.add_email(email_id=email.get("value"), is_primary=1 if email.get("primary") else 0)
+
+						if connection.get("phoneNumbers"):
+							for phone in connection.get("phoneNumbers"):
```

Comment:
you can remove instead of `if` condition
```suggestion
							for phone in connection.get("phoneNumbers", []):
```
this will save you some indentation